### ISSUE TITLE
The "julia" server crashed 5 times in the last 3 minutes.

### ISSUE THREAD

**Issue by zhiyuanzhai** [fannheyward/coc-julia#77]
I failed to start `coc-julia` after installation. Here's the log information I get: 
```
2020-12-17T22:28:27.104 DEBUG (pid:11651) [workspace] - buffer created 1
2020-12-17T22:28:27.106 DEBUG (pid:11651) [events] - Event: BufEnter [ 1 ]
2020-12-17T22:28:27.108 DEBUG (pid:11651) [events] - Event: BufWinEnter [ 1, 1000 ]
2020-12-17T22:28:28.123 WARN (pid:11651) [extensions] - Extension coc-julia activate cost more than 1s
2020-12-17T22:28:28.145 DEBUG (pid:11651) [languages] - created service source snippets
2020-12-17T22:28:28.146 DEBUG (pid:11651) [languages] - created service source snippets-source
2020-12-17T22:28:28.146 DEBUG (pid:11651) [extensions] - activate: coc-snippets
2020-12-17T22:28:28.147 INFO (pid:11651) [plugin] - coc.nvim 0.0.79-ada0b910bd initialized with node: v15.4.0 after 1074ms
2020-12-17T22:28:29.535 INFO (pid:11651) [services] - registered service "julia"
2020-12-17T22:28:29.535 DEBUG (pid:11651) [services] - starting service: julia
2020-12-17T22:28:29.536 INFO (pid:11651) [services] - Julia Language Server state change: stopped => starting
2020-12-17T22:28:29.539 INFO (pid:11651) [language-client-index] - Language server "julia" started with 11661
2020-12-17T22:28:31.252 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 21, 1 ] ]
2020-12-17T22:28:31.556 DEBUG (pid:11651) [events] - Event: CursorHold [ 1 ]
2020-12-17T22:28:43.770 INFO (pid:11651) [services] - Julia Language Server state change: starting => running
2020-12-17T22:28:43.772 DEBUG (pid:11651) [languages] - created service source julia-1
2020-12-17T22:28:43.775 INFO (pid:11651) [services] - service julia started
2020-12-17T22:28:43.775 DEBUG (pid:11651) [extensions] - activate: coc-julia
2020-12-17T22:28:46.987 INFO (pid:11651) [services] - Julia Language Server state change: running => stopped
2020-12-17T22:28:46.987 INFO (pid:11651) [services] - Julia Language Server state change: stopped => starting
2020-12-17T22:28:46.990 INFO (pid:11651) [language-client-index] - Language server "julia" started with 11663
2020-12-17T22:29:02.073 INFO (pid:11651) [services] - Julia Language Server state change: starting => running
2020-12-17T22:29:02.074 DEBUG (pid:11651) [languages] - created service source julia-1
2020-12-17T22:29:05.544 INFO (pid:11651) [services] - Julia Language Server state change: running => stopped
2020-12-17T22:29:05.544 INFO (pid:11651) [services] - Julia Language Server state change: stopped => starting
2020-12-17T22:29:05.548 INFO (pid:11651) [language-client-index] - Language server "julia" started with 11668
2020-12-17T22:29:21.693 INFO (pid:11651) [services] - Julia Language Server state change: starting => running
2020-12-17T22:29:21.693 DEBUG (pid:11651) [languages] - created service source julia-1
2020-12-17T22:29:25.328 INFO (pid:11651) [services] - Julia Language Server state change: running => stopped
2020-12-17T22:29:25.329 INFO (pid:11651) [services] - Julia Language Server state change: stopped => starting
2020-12-17T22:29:25.332 INFO (pid:11651) [language-client-index] - Language server "julia" started with 11670
2020-12-17T22:29:42.826 INFO (pid:11651) [services] - Julia Language Server state change: starting => running
2020-12-17T22:29:42.827 DEBUG (pid:11651) [languages] - created service source julia-1
2020-12-17T22:29:45.428 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 22, 1 ] ]
2020-12-17T22:29:45.732 DEBUG (pid:11651) [events] - Event: CursorHold [ 1 ]
2020-12-17T22:29:46.388 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 21, 1 ] ]
2020-12-17T22:29:46.693 DEBUG (pid:11651) [events] - Event: CursorHold [ 1 ]
2020-12-17T22:29:46.753 INFO (pid:11651) [services] - Julia Language Server state change: running => stopped
2020-12-17T22:29:46.754 INFO (pid:11651) [services] - Julia Language Server state change: stopped => starting
2020-12-17T22:29:46.758 INFO (pid:11651) [language-client-index] - Language server "julia" started with 11675
2020-12-17T22:29:46.785 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 20, 1 ] ]
2020-12-17T22:29:46.933 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 19, 1 ] ]
2020-12-17T22:29:47.098 DEBUG (pid:11651) [events] - Event: CursorMoved [ 1, [ 18, 1 ] ]
2020-12-17T22:29:47.406 DEBUG (pid:11651) [events] - Event: CursorHold [ 1 ]
2020-12-17T22:30:05.176 INFO (pid:11651) [services] - Julia Language Server state change: starting => running
2020-12-17T22:30:05.177 DEBUG (pid:11651) [languages] - created service source julia-1
2020-12-17T22:30:09.420 INFO (pid:11651) [services] - Julia Language Server state change: running => stopped
2020-12-17T22:30:25.636 DEBUG (pid:11651) [events] - Event: BufWinLeave [ 1, 1000 ]
2020-12-17T22:30:25.637 DEBUG (pid:11651) [events] - Event: BufHidden [ 1 ]
2020-12-17T22:30:25.649 DEBUG (pid:11651) [events] - Event: FileType [ 'log', 2 ]
2020-12-17T22:30:25.661 DEBUG (pid:11651) [events] - Event: BufCreate [ 2 ]
2020-12-17T22:30:25.662 DEBUG (pid:11651) [events] - Event: BufEnter [ 2 ]
2020-12-17T22:30:25.663 DEBUG (pid:11651) [events] - Event: BufWinEnter [ 2, 1000 ]
2020-12-17T22:30:25.698 DEBUG (pid:11651) [events] - Event: CursorMoved [ 2, [ 1, 1 ] ]
2020-12-17T22:30:25.707 DEBUG (pid:11651) [workspace] - buffer created 2
2020-12-17T22:30:26.004 DEBUG (pid:11651) [events] - Event: CursorHold [ 2 ]
```
It seems that the log didn't tell me what was happening... I may need help to get more information and clarity...

**Comment by zhiyuanzhai** [comment]
It's macOS 11.1 Big Sur, and I'm using Julia 1.5.3 built from MacPorts.

**Comment by zhiyuanzhai** [comment]
Problem solved by upgrading coc.nvim.
It's my being stupid here...

**Comment by zhiyuanzhai** [comment]
Well, I think the problem still exists.

**Comment by zhiyuanzhai** [comment]
To be clear, the server crashes specifically in this file (`System.jl`): 
```Julia
module System

const g = 980 # cm/s^2
const ρ = 1.0 # g/cm^3

function f(
        Du::Vector{F1},
        u::Vector{F1},
        p::Vector{Any},
        t::F2
    ) where {
        F1<:AbstractFloat, F2<:AbstractFloat,
    }

    m, a, V, Vp = p

    Du[1] = g * (m - ρ*( V(u[2])-V(u[3])) ) / (m + ρ*V(u[3]))
    Du[2] = u[1]

    ΔP = 2(a/Vp(u[3]))^2 * g*(u[2]-u[3])
    Du[3] = sign(ΔP)*sqrt(abs(ΔP))

end

end;
```
I guarantee this is a valid file. It's been tested during my last project. But I admit that the symbols are quite messy.

**Comment by fannheyward** [comment]
Can't reproduce, delete `~/.config/coc/extensions/coc-julia-data/` and try again.

**Comment by zhiyuanzhai** [comment]
Deleting `~/.config/coc/extensions/coc-julia-data/` didn't help.

But things are getting more clear now. First I moved the file away from the original working directory. The server didn't crash.   I tried again with other files in that directory. The server crashed. At this time I'm sure that there's a problem with my working directory. I looked through the subdirectories under it, discovering a `.jl` file that was actually formatted in terms of a CSV file. 

That was a result of a temporary bug in my code. I was supposing to store the data in a CSV file, but I had a typo ending the file name with an extension of `.jl`. I deleted that file and the server haven't crashed since then.
